src/components/Forms/ReviewForm.tsx

```diff
@@ -0,0 +1,181 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { useEffect, useMemo, useState } from 'react';
+import { IoStarOutline } from 'react-icons/io5';
+import { IoStar } from 'react-icons/io5';
+import { useAppDispatch, useAppSelector } from '../../redux/hooks/hooks';
+import {
+	reviewFormSchema,
+	reviewFormSchemaType,
+} from '../../validations/products/reviewFromValidations';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { createReview } from '../../redux/features/CreateReviewSlice';
+import { DynamicData } from '../../@types/DynamicData';
+//import useToast from '../../hooks/useToast';
+import { fetchReview } from '../../redux/features/getReviewSice';
+import fetchInfo from '../../utils/userDetails';
+import { editReview } from '../../redux/features/editReviewSlice';
+interface reviewFormType {
+	id: string;
+	accessToken: string;
+	handleReviewform: () => void;
+	successMessage: (message: string) => void;
+	Erromesage: (string: string) => void;
+}
+function ReviewForm(props: reviewFormType) {
+	const [isEditMode, setIsEditMode] = useState(false);
+	const [rated, setRated] = useState(0);
+	const [reviewId, setReviewId] = useState('');
+	const dispatch = useAppDispatch();
+	//const { showErrorMessage, showSuccessMessage } = useToast();
+	const {
+		register,
+		handleSubmit,
+		setValue,
+		formState: { errors },
+	} = useForm<reviewFormSchemaType>({
+		resolver: zodResolver(reviewFormSchema),
+	});
+	const data = useAppSelector(
+		(state) => state.fetchReview.data[state.fetchReview.data.length - 1]?.data,
+	);
+	const onSubmit: SubmitHandler<reviewFormSchemaType> = async (
+		data: reviewFormSchemaType,
+	) => {
+		try {
+			const { ratings, feedBack } = data;
+			const token = props.accessToken;
+			const productId = props.id;
+			if (isEditMode) {
+				console.log('EDit modeeeeeeeeeeeeeeeeeeee');
+				const res = await dispatch(
+					editReview({ reviewId, feedBack, ratings, token }),
+				).unwrap();
+				dispatch(fetchReview(props.id || '')).unwrap();
+				props.successMessage(res.message);
+			} else {
+				console.log('create modeeeeeeeeeeeeeeeeeeee');
+
+				const res = await dispatch(
+					createReview({ productId, feedBack, ratings, token }),
+				).unwrap();
+				console.log(res, 'responseeeeeeeeeeeeeeeeeeeeeee');
+				dispatch(fetchReview(props.id || '')).unwrap();
+				props.successMessage(res.message);
+			}
+
+			props.handleReviewform();
+		} catch (e) {
+			console.log('errorrrrrrrrr', e);
+			const err = e as DynamicData;
+			props.Erromesage(
+				err.data.message ||
+					err?.message ||
+					'Unknown error occured! Please try again!',
+			);
+			// showErrorMessage(
```

Comment:
Remove these line of codes you commented.

Reply:
yeah it is removed